### DISCUSSION TITLE
Bug：表格识别结果行列错乱

### DISCUSSION THREAD

**Discussion by jiataiping** [RapidAI/RapidOrientation#23] [Q&A]
系统环境：Win10 22H2
语言：Python 3.10.1
模型文件：en_ppstructure_mobile_v2_SLANet
初步原因分析：表格识别模型TableStructurer类检测到的文本框位置与RapidOCR检测到的文本框位置差别较大，导致TableMatch类执行结果行列错乱。
原始图片：
![3](https://github.com/RapidAI/RapidStructure/assets/9160007/45ee81fa-6194-48c7-94f9-cf990b72325a)
表格引擎检测到的文本框及位置：
![vis_3](https://github.com/RapidAI/RapidStructure/assets/9160007/32833915-3189-477f-9251-e5ce5a6a65cf)

**Comment by SWHL** [comment]
图中蓝色框是表格识别模型检测到的cell框（注意是cell框，表格识别模型不检测文本框），可以看到最底部的蓝色不是很准。
而RapidOCR的识别结果是很准的。下图中细线框为RapidOCR标注的，结果是比较准的。

![only_vis_det](https://github.com/RapidAI/RapidStructure/assets/28639377/b6e9707c-0754-4bc4-950d-bb259ce60cdf)

两者匹配，是以cell框为准的，因此存在错位问题。不是代码bug。
目前没有好的办法解决，除了优化微调表格识别模型。

**Comment by SWHL** [comment]
怀疑是Paddle 表格模型转换为onnx时有误差导致的。于是，我这里验证了一下直接用PaddleOCR（main分支最新代码）来推理所给原图：

推理代码（参考[link](https://github.com/PaddlePaddle/PaddleOCR/blob/main/ppstructure/table/README_ch.md#41-%E5%BF%AB%E9%80%9F%E5%BC%80%E5%A7%8B)）：
其中**1.png**即是上面所给原图
```bash
cd PaddleOCR/ppstructure

# 下载模型
mkdir inference && cd inference
# 下载PP-OCRv3文本检测模型并解压
wget https://paddleocr.bj.bcebos.com/PP-OCRv3/chinese/ch_PP-OCRv3_det_infer.tar && tar xf ch_PP-OCRv3_det_infer.tar
# 下载PP-OCRv3文本识别模型并解压
wget https://paddleocr.bj.bcebos.com/PP-OCRv3/chinese/ch_PP-OCRv3_rec_infer.tar && tar xf ch_PP-OCRv3_rec_infer.tar
# 下载PP-StructureV2中文表格识别模型并解压
wget https://paddleocr.bj.bcebos.com/ppstructure/models/slanet/ch_ppstructure_mobile_v2.0_SLANet_infer.tar && tar xf ch_ppstructure_mobile_v2.0_SLANet_infer.tar
cd ..
# 执行表格识别
python table/predict_table.py \
    --det_model_dir=inference/ch_PP-OCRv3_det_infer \
    --rec_model_dir=inference/ch_PP-OCRv3_rec_infer  \
    --table_model_dir=inference/ch_ppstructure_mobile_v2.0_SLANet_infer \
    --rec_char_dict_path=../ppocr/utils/ppocr_keys_v1.txt \
    --table_char_dict_path=../ppocr/utils/dict/table_structure_dict_ch.txt \
    --image_dir=1.png \
    --output=../output/table
```

#### 可视化结果：
![1](https://github.com/RapidAI/RapidStructure/assets/28639377/337a4abc-99c3-4a9d-a33b-1be0815d4d02)

#### Excel结果：
<img width="773" alt="image" src="https://github.com/RapidAI/RapidStructure/assets/28639377/441eb771-b674-4881-a122-cc9771fbe39b">

#### 相关附件：
[table.zip](https://github.com/user-attachments/files/15796670/table.zip)

#### 结论
1. 基于PaddleOCR表格识别结果来看，仍有错位情况，这就属于模型问题了
2. 从可视化结果来看，`rapid_table`的确与原有PaddleOCR的表格识别存在误差，后期会再次验证
